feat: add set 2 of questions for CSS animations quiz (#59790)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-css-animations/66ed8fc9f45ce3ece4053eae.md

`animation-delay`

## --quiz--

### --question--

#### --text--

What does the CSS `animation-delay` property do?

#### --distractors--

> Sets how long the animation lasts
> 
> ---
> 
> Specifies the timing function
> 
> ---
> 
> Defines animation direction
> 
> #### --answer--
> 
> Delays the start of the animation

Reviewer: ```suggestion
Sets how long the animation lasts.

---

Specifies the timing function.

---

Defines animation direction.

#### --answer--

Delays the start of the animation.
```

### --question--

#### --text--

What does `animation-fill-mode: forwards` do?

#### --distractors--

Starts the animation halfway

---

Resets the element to its original state after the animation

---

Repeats the animation indefinitely

#### --answer--

> Keeps the element at its end state after the animation

Author: Changed the question to "Which animation property specifies how the element should be styled before and after the animation?"

### --question--

#### --text--

Which value of `animation-direction` makes the animation play backward?

#### --distractors--

normal

---

alternate

---

backwards

#### --answer--

reverse

> 

Author: Here i used a different question from the lecture

"Why should CSS animations be used in moderation?"

> They automatically improve browser compatibility
> 
> ---
> 
> They increase the site’s SEO ranking
> 
> ---
> 
> They optimize server response times
> 
> #### --answer--
> 
> They can lead to poor performance and may be distracting or problematic for users with certain accessibility needs

Reviewer: right answer is twice as long as wrong answers making it easy to guess the right one.
https://contribute.freecodecamp.org/how-to-work-on-quizzes/#guidelines-for-designing-questions-and-answers

```suggestion
Too many CSS animations can lead to styles breaking as well as inconsistent styles across different browsers. 

---

Too many CSS animations can lead to lower or non existent rankings in search engine results.

---

Too many CSS animations will automatically crash the server and increase the likelihood for security risks. 

#### --answer--

Too many CSS animations can lead to poor performance and may be distracting or problematic for users with certain accessibility needs.
```

### --question--

#### --text--

Which `animation-direction` value alternates between forward and backward each cycle?

#### --distractors--

reverse

---

normal

---

backwards

#### --answer--

> alternate

Reviewer: same issue. this is not explicitly covered in the lecture. 

All the lecture says is the following:

```md
animation-direction which determines whether the animation should play forwards, backwards, or alternate.
```

It doesn't really say what the `alternate` value does. It just lists it as a possible value for the  `animation-direction` property.

I would find a way to reword this or come up with another question that is in line with the lecture

https://www.freecodecamp.org/learn/full-stack-developer/lecture-animations-and-accessibility/what-are-css-animations

Author: Changed this question to "Which animation property determines whether the animation should play forwards, backwards, or alternate?"

### --question--

#### --text--

What is the default `animation-iteration-count` if you don’t specify one?

#### --distractors--

infinite

---

0

---

none

#### --answer--

> 1

Reviewer: not specified in the lecture here

https://www.freecodecamp.org/learn/full-stack-developer/lecture-animations-and-accessibility/what-are-css-animations

The value of `1` is brought up in this lecture here, but it is not stated that is the default

https://www.freecodecamp.org/learn/full-stack-developer/lecture-animations-and-accessibility/what-are-accessibility-concerns-around-using-animations

revise question or come up with new one

Author: Used a different question here "What CSS media query detects if the user has requested minimal animations or motion effects?"

> animation-fill-mode
> 
> ---
> 
> animation-delay
> 
> ---
> 
> animation-timing-function
> 
> #### --answer--
> 
> animation-direction

Reviewer: ```suggestion
`animation-fill-mode`

---

`animation-delay`

---

`animation-timing-function`

#### --answer--

`animation-direction`
```

> What CSS media query detects if the user has requested minimal animations or motion effects?

Reviewer: ```suggestion
Which CSS media query detects if the user has requested minimal animations or motion effects?
```

> @media (reduce-motion: true)
> 
> ---
> 
> @media (min-motion-preference: low)
> 
> ---
> 
> @media (motion-preferences: reduced)
> 
> #### --answer--
> 
> @media (prefers-reduced-motion: reduce)

Reviewer: ```suggestion
`reduce-motion`

---

`min-motion-preference`

---

`motion-preferences`

#### --answer--

`prefers-reduced-motion`
```

### --question--

#### --text--

Which property sets how many times an `animation` repeats?

#### --distractors--

`animation-duration`

---

`animation-count`

---

`animation-delay`

#### --answer--

`animation-iteration-count`

### --question--

#### --text--

Which media query detects users who prefer reduced motion?

#### --distractors--

`@media (motion-preference: low)`

---

`@media (reduced-motion: true)`

---

`@media (no-motion: yes)`

#### --answer--

> `@media (prefers-reduced-motion: reduce)`

Reviewer: this is a duplicate question. replace with something else

### --question--

#### --text--

Inside the `reduced‑motion` media query, which declaration disables transitions?

#### --distractors--

`animation: none;`

---

`transition: remove;`

---

`animation-play-state: paused;`

#### --answer--

`transition: none;`

### --question--

#### --text--

What does `animation-play-state: paused` do?

#### --distractors--

Reverses the animation

---

Hides the element during animation

---

Restarts the animation

#### --answer--

> Pauses the animation

Reviewer: While the list of options does make it obvious which one to choose, I don't know if it is fair to ask this specific question.

All the lecture says is `animation-play-state which allows you to pause and resume the animation.`
There are no code examples for that nor does it explicitly show the values you can use there. So I would reword this question so it is more in line with the lecture.

Author: Reworded the question to "What does the `animation-play-state` property allow you to do?"

### --question--

#### --text--

Which unit can you use for `animation-duration` besides `s`?

#### --distractors--

`px`

---

`%`

---

`deg`

#### --answer--

> `ms`

Reviewer: not explicitly covered. come up with new question here

Author: Used a different question here "As a general rule, what content should be avoided to reduce seizure risk?"

> Set how many times the animation repeats
> 
> ---
> 
> Specify how long the animation takes to complete
> 
> ---
> 
> Determine the direction in which the animation plays
> 
> #### --answer--
> 
> Pause and resume the animation

Reviewer: ```suggestion
Set how many times the animation repeats.

---

Specify how long the animation takes to complete.

---

Determine the direction in which the animation plays.

#### --answer--

Pause and resume the animation.
```

### --question--

#### --text--

How do you apply multiple animations to one element?

#### --distractors--

Use multiple `@keyframes` rules only

---

Nest animation properties

---

Semicolon‑separate values in `animation-delay`

#### --answer--

> Comma‑separate values in `animation-name`

Reviewer: couldn't find where this was covered in the lectures or workshops. 
So we need a new question here.

Author: Used a different question here as well "Why is the !important declaration used in the prefers-reduced-motion CSS rules?"

> As a general rule, what content should be avoided to reduce seizure risk?

Reviewer: I found this question to be awkward. 

```suggestion
Which of the following is a good practice when working with animations?
```

> Any content that changes color
> 
> ---
> 
> Any movement of elements
> 
> ---
> 
> Any use of the transform property
> 
> #### --answer--
> 
> Content that flashes more than three times per second

Reviewer: ```suggestion
Use as many flashing colors and rapid movements as possible to grab attention.

---

Avoid testing animations on different devices or screen sizes.

---

Make animations last as long as possible to ensure users notice them.

#### --answer--

Avoid content that flashes more than three times per second to prevent triggering seizures or causing discomfort.
```

### --question--

#### --text--

In `animation: slide 2s ease-in 1s infinite alternate paused;`, what does `alternate` specify?

#### --distractors--

The timing function

---

The fill mode

---

The delay before starting

#### --answer--

> That direction alternates each iteration

Reviewer: this doesn't seem to be explicitly covered either. So I would reword this one too so it matches the lectures and workshops content.

Author: Used a different question from the lesson "What does `animation-iteration-count: 1 !important;` ensure in CSS?"

### --question--

#### --text--

Which of these is a valid `animation-fill-mode` value?

#### --distractors--

repeat

---

loop

---

cycle

#### --answer--

> both

Reviewer: options not covered in lecture. come up with new question here

Author: Used a different question here "What type of animation content can trigger seizures in people with photosensitive epilepsy?"

> To prevent other media queries from loading
> 
> ---
> 
> To limit styles to the first child element
> 
> ---
> 
> To debug CSS more easily
> 
> #### --answer--
> 
> To ensure these rules take precedence over other styles

Reviewer: ```suggestion
To prevent other media queries from loading.

---

To limit styles to the first child element.

---

To debug CSS more easily.

#### --answer--

To ensure these rules take precedence over other styles.
```

> That animations are paused
> 
> ---
> 
> That animations run infinitely
> 
> ---
> 
> That animations reverse direction each cycle
> 
> #### --answer--
> 
> That any looping animations only play once

Reviewer: ```suggestion
That animations are paused.

---

That animations run infinitely.

---

That animations reverse direction each cycle.

#### --answer--

That any looping animations only play once.
```

> Rapid flashing or strobing effects that flash more than three times per second

Reviewer: this question is very similar to an earlier one. replace it with something else

Author: Replaced this with "What CSS property is used to specify how long an animation should take to complete?"

But there is already a somewhat similar question: "Which CSS property is used to specify that an animation should take 5 seconds to complete?"

What are your thoughts on this?

Reviewer: I think it is fine

### --question--

#### --text--

Which property is NOT part of the `animation` shorthand?

#### --distractors--

animation-delay

---

animation-timing-function

---

animation-direction

#### --answer--

animation-transform

### --question--

#### --text--

What does the `@keyframes` rule define?

#### --distractors--

The timing function of an animation

---

The default state of an element

---

The media queries for animations

#### --answer--

The sequence of styles at different points in an animation

### --question--

#### --text--

Which property is used for simple two‑state changes instead of a full `animation`?

#### --distractors--

`animation`

---

`transform`

---

`@keyframes`

#### --answer--

> `transition`

Reviewer: couldn't find where this was explicitly discussed. so we will need to come up with a new question here.

Author: Used this question here "What is the goal of using the `prefers-reduced-motion` media query in animations?"

> animation-delay
> 
> ---
> 
> animation-timing-function
> 
> ---
> 
> animation-direction
> 
> #### --answer--
> 
> animation-transform

Reviewer: ```suggestion
`animation-delay`

---

`animation-timing-function`

---

`animation-direction`

#### --answer--

`animation-transform`
```

> The timing function of an animation
> 
> ---
> 
> The default state of an element
> 
> ---
> 
> The media queries for animations
> 
> #### --answer--
> 
> The sequence of styles at different points in an animation

Reviewer: ```suggestion
The timing function of an animation.

---

The default state of an element.

---

The media queries for animations.

#### --answer--

The sequence of styles at different points in an animation.
```

### --question--

#### --text--

In a keyframes rule, what does `100%` represent?

#### --distractors--

The start of the animation

---

The halfway point

---

The easing function

#### --answer--

The end of the animation

### --question--

#### --text--

Which property controls the pace of an `animation` over its duration?

#### --distractors--

`animation-duration`

---

`animation-delay`

---

`animation-iteration-count`

#### --answer--

`animation-timing-function`

### --question--

#### --text--

What does `animation: none` do?

#### --distractors--

Sets animation duration to default

---

Resets animation direction to normal

---

Pauses the animation

#### --answer--

> Disables all animations on the element

Reviewer: couldn't find where this was specifically taught. so we should revise this question.

Author: Used a new question here "What should developers consider when implementing animations to maintain accessibility?"

> Use a low-motion approach that respects user settings for comfortable accessibility.

Reviewer: this question is pretty similar to a few other questions from earlier. come up with different one

> The start of the animation
> 
> ---
> 
> The halfway point
> 
> ---
> 
> The easing function
> 
> #### --answer--
> 
> The end of the animation

Reviewer: ```suggestion
The start of the animation.

---

The halfway point.

---

The easing function.

#### --answer--

The end of the animation.
```

> Use purposeful animation, avoid sudden large motions, provide controls, and implement `prefers-reduced-motion`

Reviewer: sentences need periods and the correct answer should not be twice as long as wrong ones. Please make the appropriate updates. 
https://contribute.freecodecamp.org/how-to-work-on-quizzes/#guidelines-for-designing-questions-and-answers

### --question--

#### --text--

Which of the following is the correct syntax to slide an element in from the left?

#### --distractors--

```js
@keyframes slide-in {
  0    { transform: translate(-100%); }
  100  { transform: translate(0); }
}
```

---

```js
@keyframes slide-in {
  from { translateX(-100%); }
  to   { translateX(0); }
}
```

---

```js
@keyframes slide-in {
  start { transform: moveX(-100%); }
  end   { transform: moveX(0); }
}
```

#### --answer--

```js
@keyframes slide-in {
  0%   { transform: translateX(-100%); }
  100% { transform: translateX(0); }
}

> ```

Reviewer: the wrong language is specified here. it should be css not js.
Also, css examples should use formatting consistent with review pages and lectures like this

```css
@keyframes slide-in {
  0% {
    transform: translateX(-100%);
  }
  100% {
    transform: translateX(0);
  }
}

```

please fix all options
